### PR TITLE
set-log-severity-for-managed-cluster-components

### PR DESCRIPTION
**Pull request by AdamSadek**
Enhancement for: https://github.com/projectsveltos/sveltosctl/issues/41

### Description
Enhanced the sveltosctl log-settings command to allow users to set log severity levels for components running in managed clusters. The user must specify the cluster namespace and name to apply the changes.

### Changes
1. Updated log-settings command to support managed cluster components.
2. Modified logic to request cluster namespace and name from the user.
3. Added necessary error handling and parameter validation.

### Testing
I ensured I can see the cluster name after setting the log-level set to contain the cluster name
```
┌──(sdk㉿sdk)-[~/sveltos/sveltosctl]
└─$ ./bin/sveltosctl log-level set --component=AddonManager --namespace=namespace-test --cluster=cluster-name-test --debug
                                                                                                                                           
┌──(sdk㉿sdk)-[~/sveltos/sveltosctl]
└─$ cd ../addon-controller                                                                                                 
                                                                                                                                           
┌──(sdk㉿sdk)-[~/sveltos/addon-controller]
└─$ export KUBECONFIG=$(pwd)/test/fv/workload_kubeconfig && kubectl get debuggingconfiguration -o yaml                     

apiVersion: v1
items:
- apiVersion: lib.projectsveltos.io/v1alpha1
  kind: DebuggingConfiguration
  metadata:
    creationTimestamp: "2024-05-21T11:26:08Z"
    generation: 1
    managedFields:
    - apiVersion: lib.projectsveltos.io/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        f:spec:
          .: {}
          f:configuration: {}
      manager: sveltosctl
      operation: Update
      time: "2024-05-21T11:26:08Z"
    name: cluster-name-test
    resourceVersion: "1784"
    uid: 8fbd859c-2371-4711-a8ac-135c6f6a745f
  spec:
    configuration:
    - component: AddonManager
      logLevel: LogLevelDebug
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""

```

### CODE REVIEW

File: internal/commands/loglevel/set.go [hunk 1]
```diff
@@ -62,22 +62,24 @@ func updateDebuggingConfiguration(ctx context.Context, logSeverity libsveltosv1a
...
 
-	return updateLogLevelConfiguration(ctx, spec)
+	return updateLogLevelConfiguration(ctx, spec, namespace, clusterName)
 }
 
 // Set displays/changes log verbosity for a given component
 func Set(ctx context.Context, args []string) error {
 	doc := `Usage:
-  sveltosctl log-level set --component=<name> (--info|--debug|--verbose)
+  sveltosctl log-level set --component=<name> --namespace=<namespace> --cluster=<cluster-name> (--info|--debug|--verbose)
```
Comment (gianlucam76) [inline internal/commands/loglevel/set.go:71]: we need the cluster-type as well, which can either be Capi or Sveltos

Capi indicates clusterAPI powered cluster
Sveltos indicates a cluster registered with Sveltos (essentially everything else but ClusterAPI powered cluster).

Comment (gianlucam76) [inline internal/commands/loglevel/set.go:71]: namespace and cluster are optional (and cluster-type as well).

So those need to be 

[--namespace=<namespace>] [--cluster=<cluster-name>]

File: internal/commands/loglevel/set.go [hunk 2]
```diff
@@ -62,22 +62,24 @@ func updateDebuggingConfiguration(ctx context.Context, logSeverity libsveltosv1a
...
 Options:
   -h --help             Show this screen.
      --component=<name> Name of the component for which log severity is being set.
```
Comment (gianlucam76) [inline internal/commands/loglevel/set.go:74]: nit: can you please make sure indentation for the comment is same for all options? it makes it easier to read when doing

sveltosctl low-level set --help

File: internal/commands/loglevel/show.go [hunk 3]
```diff
@@ -52,12 +52,14 @@ func showLogSettings(ctx context.Context) error {
 // Show displays information about log verbosity (if set)
 func Show(ctx context.Context, args []string) error {
 	doc := `Usage:
-  sveltosctl log-level show
+  sveltosctl log-level show --namespace=<namespace> --cluster=<cluster-name>
```
Comment (gianlucam76) [inline internal/commands/loglevel/show.go:55]: same comments as above:

1. [--namespace=<namespace>] [--cluster=<cluster-name>] cause those are optional
2. add cluster-type
3. indentation for the comment of each option should be same

File: internal/commands/loglevel/show_test.go [hunk 4]
```diff
@@ -52,7 +59,7 @@ var _ = Describe("Show", func() {
 		c := fake.NewClientBuilder().WithScheme(scheme).WithObjects(initObjects...).Build()
 
 		utils.InitalizeManagementClusterAcces(scheme, nil, nil, c)
-		err = loglevel.ShowLogSettings(context.TODO())
+		err = loglevel.ShowLogSettings(context.TODO(), testNamespace, testClusterName)
```
Comment (gianlucam76) [inline internal/commands/loglevel/show_test.go:62]: loglevel.ShowLogSettings(context.TODO(), "", "") so we can test when looking at the management cluster it works and

loglevel.ShowLogSettings(context.TODO(), testNamespace, testClusterName) to verify in a managed cluster

File: internal/commands/loglevel/unset.go [hunk 5]
```diff
@@ -50,21 +49,23 @@ func unsetDebuggingConfiguration(ctx context.Context, component string) error {
...
+		return updateLogLevelConfiguration(ctx, spec, namespace, clusterName)
 	}
 	return nil
 }
 
 // Unset resets log verbosity for a given component
 func Unset(ctx context.Context, args []string) error {
 	doc := `Usage:
-  sveltosctl log-level unset --component=<name>
+  sveltosctl log-level unset --component=<name> --namespace=<namespace> --cluster=<cluster-name>
```
Comment (gianlucam76) [inline internal/commands/loglevel/unset.go:60]: same comments as above:

[--namespace=] [--cluster=] cause those are optional
add cluster-type
indentation for the comment of each option should be same

File: internal/utils/debuggingconfigurations.go [hunk 6]
```diff
@@ -29,15 +29,18 @@ const (
...
 	ctx context.Context,
+	namespace string,
+	clusterName string,
 ) (*libsveltosv1alpha1.DebuggingConfiguration, error) {
 
 	req := &libsveltosv1alpha1.DebuggingConfiguration{}
 
 	reqName := client.ObjectKey{
-		Name: defaultInstanceName,
+		Namespace: namespace,
```
Comment (gianlucam76) [inline internal/utils/debuggingconfigurations.go:42]: here we need to distinguish two scenarios. 

If namespace is empty (and so clusterName and clusterType) we can use a.client (which points to the management cluster),

otherwise we need to [clusterproxy.GetSveltosKubernetesClient](https://github.com/projectsveltos/libsveltos/blob/fd9879699f8d4e43b1412ffcca3e253525d70a6d/lib/clusterproxy/clusterproxy.go#L142) to get the client pointing to the managed cluster.

**Review by gianlucam76 — COMMENTED** [review]
Great job @AdamSadek 

I added few comments but PR is a great start!

File: internal/commands/loglevel/show_test.go (hunk 4 above)
Comment (AdamSadek) [reply internal/commands/loglevel/show_test.go:62]: Should I also add the _clusterType_ to the tests or keep it at those two?

Comment (gianlucam76) [reply internal/commands/loglevel/show_test.go:62]: We need ShowLogSettings to accept clusterType as well. But for testing we can pass empty values.

File: internal/utils/debuggingconfigurations.go (hunk 6 above)
Comment (AdamSadek) [reply internal/utils/debuggingconfigurations.go:51]: Should we also pass clusterType to GetSveltosKubernetesClient?

Comment (gianlucam76) [reply internal/utils/debuggingconfigurations.go]: sorry my bad. This is the [method](https://github.com/projectsveltos/libsveltos/blob/fd9879699f8d4e43b1412ffcca3e253525d70a6d/lib/clusterproxy/cluster_utils.go#L203) to use and that requires clusterType.

File: internal/commands/loglevel/set_test.go [hunk 7]
```diff
@@ -17,46 +17,55 @@ limitations under the License.
...
+        
+        dc := &libsveltosv1alpha1.DebuggingConfiguration{
+            ObjectMeta: metav1.ObjectMeta{
+                Name: utils.DefaultInstanceName,
+            },
+        }
+        err = c.Create(context.TODO(), dc)
+        Expect(err).To(BeNil())
+
+        Expect(loglevel.UpdateDebuggingConfiguration(context.TODO(), libsveltosv1alpha1.LogLevelDebug,
```
Comment (gianlucam76) [inline internal/commands/loglevel/set_test.go:48]: the only change you need to make to those tests is to add extra argument as empty
